### PR TITLE
Correct handling of 'special' dependencies in poms

### PR DESCRIPTION
**Pull request by mockitoguy**
Dependencies declared with target configurations need extra work to get them correctly mapped to pom files. Pom files are lossy when compared to Gradle or Ivy model of dependencies.

Added 'all' module that enables customers to declare a single dependency and get all directly consumable Gobblin artifacts.

Tested by running './gradlew publishToMavenLocal' and inspecting the build output and the generated pom files.

### CODE REVIEW

**Comment by mockitoguy** [comment]
Travis CI failure is due to test flakiness.

File: gobblin-all/build.gradle [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+apply plugin: "java"
+
+dependencies {
+    //TODO: list all dependencies that are directly consumable
+    compile project(":gobblin-api")
+    compile project(":gobblin-rest-service:gobblin-rest-api")
+    compile project(path: ':gobblin-rest-service:gobblin-rest-api', configuration: 'restClient')
+    compile project(path: ':gobblin-rest-service:gobblin-rest-api', configuration: 'dataTemplate')
+}
```
Comment (vikrambohra) [inline gobblin-all/build.gradle:12]: Please add all gobblin modules provided on slack

File: gradle/scripts/bintrayPublishing.gradle [hunk 2]
```diff
@@ -52,6 +52,42 @@ subprojects{
...
+                def newArtifactId = d.name + "-" + confToPom[d.targetConfiguration]
+                dependencyNode.appendNode('artifactId', newArtifactId)
+                dependencyNode.appendNode('version', d.version)
+                added.add(newArtifactId)
+              }
+            }
+            if (added || removed) {
+              logger.lifecycle("Updated pom dependencies in {}, added: {}, removed: {}", project.path, added, removed)
+            }
+          }
```
Comment (vikrambohra) [inline gradle/scripts/bintrayPublishing.gradle:90]: Thanks for this. It looks clean. Can you verify if the generated pom also adds global dependency exclusions? (scripts/gradle/globalDependencies.gradle)

Comment (vikrambohra) [reply gradle/scripts/bintrayPublishing.gradle:100]: Based on spot check. Looks like the dependency exclusion rules are only applied to the newly created (gobblin-rest-api-rest-client etc) dependencies in the POM. But does the initial POM have the dependency exclusion rules as well?

Comment (mockitoguy) [reply gradle/scripts/bintrayPublishing.gradle:100]: They are applied to all dependencies: https://gist.github.com/mockitoguy/662dc4d0f444ee082c3ccc65d9526f35

**Pull request closed without merging**